sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -191,7 +209,45 @@ def plot(self, plot_method="contourf", ax=None, xlabel=None, ylabel=None, **kwar
             _, ax = plt.subplots()
 
         plot_func = getattr(ax, plot_method)
-        self.surface_ = plot_func(self.xx0, self.xx1, self.response, **kwargs)
+        if self.response.ndim == 2:
+            self.surface_ = plot_func(self.xx0, self.xx1, self.response, **kwargs)
+        else:  # self.response.ndim == 3
+            colors = self.multiclass_colors
+            multiclass_cmaps = []
+            if isinstance(colors, str) or colors is None:
+                cmap = "viridis" if colors is None else colors
+                cmap = mpl.colormaps[cmap].resampled(self.response.shape[-1])
+                for class_idx, primary_color in enumerate(cmap.colors):
+                    r, g, b, _ = primary_color
+                    class_cmap = mpl.colors.LinearSegmentedColormap.from_list(
+                        f"colormap_{class_idx}", [(1.0, 1.0, 1.0, 1.0), (r, g, b, 1.0)]
+                    )
+                    multiclass_cmaps.append(class_cmap)
+            else:
+                for class_idx, class_color in enumerate(colors):
+                    r, g, b, _ = mpl.colors.to_rgba(class_color)
+                    class_cmap = mpl.colors.LinearSegmentedColormap.from_list(
+                        f"colormap_{class_idx}", [(1.0, 1.0, 1.0, 1.0), (r, g, b, 1.0)]
+                    )
+                    multiclass_cmaps.append(class_cmap)
+
+            self.surface_ = []
+            for class_idx, cmap in enumerate(multiclass_cmaps):
+                response = np.ma.array(
+                    self.response[:, :, class_idx],
+                    mask=~(self.response.argmax(axis=2) == class_idx),
+                )
+                # `cmap` should not be in kwargs
+                if "cmap" in kwargs:
+                    del kwargs["cmap"]
```

Comment:
Not 100% I should be deleting a `kwarg` here...

Reply:
Yep this might be safer to raise an error instead mentioning that you cannot pass a `cmap` parameter and instead pass `multiclass_colors`.

Reply:
Otherwise we could create a new dictionary that just remove cmap and raise the warning but you are right that we should not modify the original dictionary.

Reply:
I make a copy of the dict, `safe_kwargs`. Happy to change though!